components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authz/handlers/AbstractResponseTypeHandlerTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.authz.handlers;
+
+import org.apache.oltu.oauth2.common.message.types.ResponseType;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AuthorizeReqDTO;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AuthorizeRespDTO;
+
+import java.io.File;
+import java.lang.reflect.Field;
+
+@PrepareForTest({IdentityUtil.class})
+@PowerMockIgnore({"javax.net.*", "javax.security.*", "javax.crypto.*"})
+public class AbstractResponseTypeHandlerTest extends PowerMockTestCase {
+    AbstractResponseTypeHandler abstractResponseTypeHandler;
+
+    @BeforeClass
+    public void setUp() throws Exception {
+        abstractResponseTypeHandler = new AbstractResponseTypeHandler() {
+            @Override
+            public OAuth2AuthorizeRespDTO
+            issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx)
+                    throws IdentityOAuth2Exception {
+                return null;
+            }
+        };
+
+        System.setProperty("carbon.home", System.getProperty("user.dir"));
```

Comment:
Let's add target directory. In a case if something / some file is created at run time, those things will be deleted.